contracts/ColonyTask.sol

```diff
@@ -292,27 +294,18 @@ contract ColonyTask is ColonyStorage, DSMath {
   taskNotFinalized(_id)
   {
     Task storage task = tasks[_id];
-    Role storage managerRole = task.roles[MANAGER];
-    Role storage workerRole = task.roles[WORKER];
-
     task.finalized = true;
-
-    IColonyNetwork colonyNetworkContract = IColonyNetwork(colonyNetworkAddress);
     uint skillId = task.skills[0];
 
-    uint managerPayout = task.payouts[MANAGER][token];
-    uint workerPayout = task.payouts[WORKER][token];
-
     // NOTE: Ratings are already 10 multiplied because of the requirement to support 0.5 subtraction of rating values
     // NOTE: reputation change amount is hereby limited to MAXINT/50
-    int managerReputation = SafeMath.mulInt(int(managerPayout), (int(managerRole.rating)*2 - 50)) / 50;
-    //TODO: Change the skillId to represent the task domain
-    colonyNetworkContract.appendReputationUpdateLog(managerRole.user, managerReputation, skillId);
+    updateReputation(_id, MANAGER, skillId);
 
     // NOTE: reputation change amount is hereby limited to MAXINT/30
-    int workerReputation = SafeMath.mulInt(int(workerPayout), (int(workerRole.rating)*2 - 50)) / 30;
-    colonyNetworkContract.appendReputationUpdateLog(workerRole.user, workerReputation, skillId);
+    updateReputation(_id, WORKER, skillId);
     // TODO Reputation changes for other relevant roles, domains.
+
+    updateReputation(_id, EVALUATOR, skillId);
```

Comment:
I don't see the benefit of calling another function 3 times from here. This incurs some unnecessary extra reads from storage. We can refactor out the internal function in cases where the logic lies in multiple functions but here it's all within `finalizeTask`. Plus with the suggestion from above of balancing gas costs and design you best keep it all in this function.

Reply:
A small note here to switch the order in which `appendReputationUpdateLog` is called to match the order of role ids, i.e. manager (roleId = 0), evaluator (roleId = 1), worker (roleId = 2). Just so when getting the respective reputation log entries, it's more intuitive.

Reply:
I was getting the 'stack too deep' error as i mentioned in description, because of too much local variables (16 is max i think).

I will try to bring back everything into finalizeTask, but i will probably remove some local variables like [this one](https://github.com/JoinColony/colonyNetwork/blob/develop/contracts/ColonyTask.sol#L303) or [this one](https://github.com/JoinColony/colonyNetwork/blob/develop/contracts/ColonyTask.sol#L296) and access them directly [here](https://github.com/JoinColony/colonyNetwork/blob/develop/contracts/ColonyTask.sol#L308).
What do you think?

Reply:
I had problems when I tried to do that if I remember correctly. But otherwise yes, try cutting down some of the local variables.

Reply:
`skillId` does not need to be explicit as well.